### PR TITLE
fix qt unit test

### PR DESCRIPTION
**Pull request by miyazakh**
This PR is to fix qt unit cases.

fix qt unit test verifyClientCertificate
fix error code when being opensslextra
compliant with OpenSSL at verify_locations
set usePSK at the beginning

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -1879,6 +1879,14 @@ int InitSSL_Ctx(WOLFSSL_CTX* ctx, WOLFSSL_METHOD* method, void* heap)
     ctx->noPskDheKe = 1;
 #endif
 
+#if defined(WOLFSSL_QT)
```
Comment (cconlon) [inline src/internal.c:1882]: Does this also need `&& !defined(NO_PSK)` to make sure PSK is enabled?

File: src/internal.c [hunk 2]
```diff
@@ -1879,6 +1879,14 @@ int InitSSL_Ctx(WOLFSSL_CTX* ctx, WOLFSSL_METHOD* method, void* heap)
...
+    /* Qt retrieves supported cipher list at initialization 
+     * from get_cipher_compat().
+     * Qt doesn't not allow to use a cipher if it is not in the supported list.
```
Comment (cconlon) [inline src/internal.c:1885]: Let's re-word this to "Qt doesn't allow use of a cipher if it is not in the supported list."

File: src/ssl.c [hunk 3]
```diff
@@ -7027,12 +7027,31 @@ int wolfSSL_CTX_load_verify_locations_ex(WOLFSSL_CTX* ctx, const char* file,
...
             ret = fileRet;
+    #if defined(WOLFSSL_QT)
+            if (ret == BAD_PATH_ERROR && 
+                flags & WOLFSSL_LOAD_FLAG_IGNORE_BAD_PATH_ERR) {
+           /* QSslSocket always loads certs in system folder 
+            * when it is initialized.
+            * Compliant with OpenSSL when flag sets.*/
+                ret = WOLFSSL_SUCCESS;
+            }
+            else
```
Comment (cconlon) [inline src/ssl.c:7038]: Please add brackets around else body here.

File: src/ssl.c [hunk 4]
```diff
@@ -18845,6 +18868,13 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_set_peer_cert_chain(WOLFSSL* ssl)
     if (sk == NULL) {
         WOLFSSL_MSG("Null session chain");
     }
+#if defined(WOLFSSL_QT)
+    else if (ssl->options.side == WOLFSSL_SERVER_END) {
```
Comment (cconlon) [inline src/ssl.c:18872]: Is this Qt-specific, or just OpenSSL specific?  I'm wondering if this should be protected by a define more generic than WOLFSSL_QT.  OPENSSL_EXTRA/OPENSSL_ALL?

**Review by cconlon — CHANGES_REQUESTED** [review]

File: src/internal.c (hunk 1 above)
Comment (miyazakh) [reply src/internal.c:1882]: You are correct. Should be protected with !defined(NO_PSK)

File: src/internal.c (hunk 2 above)
Comment (miyazakh) [reply src/internal.c:1885]: Thanks!

File: src/ssl.c (hunk 3 above)
Comment (miyazakh) [reply src/ssl.c:7038]: Add brackets and moved comments above in 4 spaces

File: src/ssl.c (hunk 4 above)
Comment (miyazakh) [reply src/ssl.c:18872]: This is OpenSSL specific, and this is for server side only. When we look at tls_process_server_certificate(), the comments say
"Inconsistency alert: cert_chain does include the peer's certificate, which we don't include in statem_srvr.c". This means client  side chain includes peer's certificate, but not server side.

Changed guard from WOLFSSL_QT to OPENSSL_ALL

**Review by cconlon — APPROVED** [review]

**Pull request merged**